soc/espressif/esp32s3/esp32s3-mp.c

```diff
@@ -35,7 +38,135 @@ void esp_appcpu_start(void *entry_point)
 
 	esp_rom_ets_set_appcpu_boot_addr((void *)entry_point);
 
-	ets_delay_us(50000);
+	esp_cpu_reset(1);
+}
+
+static int load_segment(uint32_t src_addr, uint32_t src_len, uint32_t dst_addr)
+{
+	const uint32_t *data = (const uint32_t *)sys_mmap(src_addr, src_len);
+
+	if (!data) {
+		ets_printf("%s: mmap failed", __func__);
+		return -1;
+	}
+
+	volatile uint32_t *dst = (volatile uint32_t *)dst_addr;
+
+	for (int i = 0; i < src_len / 4; i++) {
+		dst[i] = data[i];
+	}
+
+	sys_munmap(data);
+
+	return 0;
+}
+
+int IRAM_ATTR esp_appcpu_image_load(unsigned int hdr_offset, unsigned int *entry_addr)
+{
+	const uint32_t img_off = FIXED_PARTITION_OFFSET(slot0_appcpu_partition);
+	const uint32_t fa_size = FIXED_PARTITION_SIZE(slot0_appcpu_partition);
+	const uint8_t fa_id = FIXED_PARTITION_ID(slot0_appcpu_partition);
+	int rc = 0;
+
```

Comment:
Maybe same here, we can do even better, before loading the image, disable both I and D cache to avoid cache coherency issues.